common-web-app-guide.md

```diff
@@ -0,0 +1,146 @@
+# ODL Docker/Django/Webpack Web App Guide
+
+**SECTIONS**
+1. [Initial Setup](#initial-setup)
+2. [Running and Accessing the App](#running-and-accessing-the-app)
+3. [Troubleshooting](#troubleshooting)
+
+
+# Initial Setup
+
+### Major Dependencies
+- Docker
+  - _[OSX only]_ Recommended OSX install method: [Download from Docker website](https://docs.docker.com/mac/)
+- docker-compose
+  - Recommended install: pip (`pip install docker-compose`)
+- _[OSX only]_
+  - Virtualbox (https://www.virtualbox.org/wiki/Downloads)
+  - Node/NPM 
+    - Recommended install method: [Installer on Node website](https://nodejs.org/en/download/)
+    - *NOTE: No specific versions have been chosen yet for NPM.*
+  - Yarn
+    - Some of our projects ship with a helpful yarn installation script: `sudo node ./scripts/install_yarn.js`
+    - If the the install script doesn't exist, install by running: `npm install -g yarn@0.27.0` (or whatever version is specified for yarn in `package.json`; use `sudo` as necessary)
+
+All commands in this guide should be run from the root directory of your project's repository (unless specified otherwise).
+
+### _[OSX only]_ Create a Docker VM for this app
+
+Replace `MACHINENAME` with a name that indicates the project (e.g.: `mm` or `micromasters` for the MicroMasters project). Allowed characters for the machine name: alphanumeric characters, periods, and dashes.
+
+    docker-machine create --driver virtualbox MACHINENAME
+    docker-machine start MACHINENAME
+    # 'docker-machine env MACHINENAME' prints export commands for important environment variables
+    eval "$(docker-machine env MACHINENAME)"
```

Comment:
docker-machine has it's faults for sure, but it is the standard for all OSX users on the team. I don't think we've dedicated the time to really evaluate Docker on OSX sans docker-machine (though maybe we have and I'm just not aware of it). Until we do that, i think the instructions should remain as they are